rethinkdb/datadog_checks/rethinkdb/check.py

```diff
@@ -0,0 +1,108 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from contextlib import contextmanager
+from typing import Any, Callable, Iterator, Sequence, cast
+
+import rethinkdb
+
+from datadog_checks.base import AgentCheck
+
+from . import operations, queries
+from .config import Config
+from .document_db import DocumentQuery
+from .document_db.types import Metric
+from .types import Instance
+
+
+class RethinkDBCheck(AgentCheck):
+    """
+    Collect metrics from a RethinkDB cluster.
+    """
+
+    SERVICE_CHECK_CONNECT = 'rethinkdb.can_connect'
+
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Any) -> None
+        super(RethinkDBCheck, self).__init__(*args, **kwargs)
+        self.config = Config(cast(Instance, self.instance))
+        self.queries = (
+            queries.config_summary,
+            queries.cluster_statistics,
+            queries.server_statistics,
+            queries.table_statistics,
+            queries.replica_statistics,
+            queries.table_statuses,
+            queries.server_statuses,
+            queries.system_jobs,
+            queries.current_issues_summary,
+        )  # type: Sequence[DocumentQuery]
+
+    @contextmanager
+    def connect_submitting_service_checks(self):
+        # type: () -> Iterator[rethinkdb.net.Connection]
+        config = self.config
+
+        tags = ['host:{}'.format(config.host), 'port:{}'.format(config.port)]
+        tags.extend(config.tags)
+
+        try:
+            with rethinkdb.r.connect(
+                host=config.host,
+                port=config.port,
+                user=config.user,
+                password=config.password,
+                ssl={'ca_certs': config.tls_ca_cert} if config.tls_ca_cert is not None else {},
+            ) as conn:
+                yield conn
+        except rethinkdb.errors.ReqlDriverError as exc:
+            message = 'Could not connect to RethinkDB server: {!r}'.format(exc)
+            self.log.error(message)
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, tags=tags, message=message)
+            raise
+        except Exception as exc:
+            message = 'Unexpected error while executing RethinkDB check: {!r}'.format(exc)
+            self.log.error(message)
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, tags=tags, message=message)
+            raise
```

Comment:
Does password ever get exposed in errors?

Reply:
Not that I know of (hasn't happened during development at least), but I get where you're coming from. Do we have some sort of generic password filtering util?

Reply:
Filed https://github.com/DataDog/integrations-core/pull/6107

Reply:
```json
  {
    "check": "rethinkdb.can_connect",
    "host_name": "docker-desktop",
    "timestamp": 1585129926,
    "status": 0,
    "message": "Hey, leak this password: '********'",
    "tags": [
      "env:testing",
      "host:localhost",
      "port:28015"
    ]
  }
```

✅